### PR TITLE
Adding status to ingress reports

### PR DESCRIPTION
**Pull request by lcouzens**
## Jira Ticket

[COST-3544](https://issues.redhat.com/browse/COST-3544)

## Description

This change will add a status field to the posted ingress reports so we can update them on failed file processing.

## Testing

1. Checkout Branch
2. Restart Koku
3. Add a storage only source AWS/Azure/GCP
4. Post a report with Missing Required columns
5. We should gracefully bomb out + update the status field on the posted report

## Notes

...

### CODE REVIEW

**Comment by lcouzens** [comment]
Smoke tests passed

File: koku/masu/database/ingress_report_db_accessor.py [hunk 1]
```diff
@@ -48,4 +48,18 @@ def mark_ingress_report_as_completed(self, ingress_report_uuid):
...
+
+    def update_ingress_report_status(self, ingress_report_uuid, status):
+        """Update the status field for ingress reports."""
+        if ingress_report_uuid:
+            ingress_report = self._get_db_obj_query().filter(uuid=ingress_report_uuid).first()
+            ingress_report.status = status
+            ingress_report.save()
+            msg = (
+                f"Updating ingress report {ingress_report.uuid} "
+                f"\nfor provider {ingress_report.source} "
```
Comment (chambridge) [inline koku/masu/database/ingress_report_db_accessor.py:62]:
```suggestion
                f"\nfor source {ingress_report.source} "
```

File: koku/masu/processor/parquet/ocp_cloud_parquet_report_processor.py [hunk 2]
```diff
@@ -211,22 +211,22 @@ def get_ocp_provider_uuids(self):
...
-                    openshift_filtered_data_frame = self.ocp_on_cloud_data_processor(
-                        daily_data_frame, cluster_topology, matched_tags
-                    )
-
-                    if self.provider_type in (Provider.PROVIDER_GCP, Provider.PROVIDER_GCP_LOCAL):
-                        self.create_partitioned_ocp_on_cloud_parquet(
-                            openshift_filtered_data_frame, parquet_base_filename, i
+        if daily_data_frames != []:
+            ocp_provider_uuids = self.get_ocp_provider_uuids()
+            # # Get OpenShift topology data
```
Comment (chambridge) [inline koku/masu/processor/parquet/ocp_cloud_parquet_report_processor.py:216]:
```suggestion
            # Get OpenShift topology data
```

File: koku/api/ingress/reports/serializers.py [hunk 3]
```diff
@@ -26,8 +26,10 @@ class Meta:
             "completed_timestamp",
             "reports_list",
             "source",
+            "source_id",
             "bill_year",
             "bill_month",
+            "status",
```
Comment (chambridge) [inline koku/api/ingress/reports/serializers.py:32]: I feel like some of these fields should be marked as `read-only=true` right?

https://www.django-rest-framework.org/community/3.0-announcement/#the-extra_kwargs-option

Seems like the uuid, timestamp, and status fields would be, right?

**Review by chambridge — CHANGES_REQUESTED** [review]
Minor comments.

File: koku/masu/processor/parquet/ocp_cloud_parquet_report_processor.py (hunk 2 above)
Comment (lcouzens) [reply koku/masu/processor/parquet/ocp_cloud_parquet_report_processor.py:216]: Thats gotta be the best mistake from a merge conflict resolution to be fair :rofl:

**Review by chambridge — APPROVED** [review]
LGTM

**Pull request merged**